pkg/virt-operator/install-strategy/create.go

```diff
@@ -731,6 +731,61 @@ func createOrUpdateConfigMaps(kv *v1.KubeVirt,
 
 	return nil
 }
+func rolloutNonCompatibleCRDChanges(kv *v1.KubeVirt,
+	targetStrategy *InstallStrategy,
+	stores util.Stores,
+	clientset kubecli.KubevirtClient,
+	expectations *util.Expectations) error {
+
+	ext := clientset.ExtensionsClient()
+
+	version := kv.Status.TargetKubeVirtVersion
+	imageRegistry := kv.Status.TargetKubeVirtRegistry
+	id := kv.Status.TargetDeploymentID
+
+	for _, crd := range targetStrategy.crds {
+		var cachedCrd *extv1beta1.CustomResourceDefinition
+
+		crd := crd.DeepCopy()
+		obj, exists, _ := stores.CrdCache.Get(crd)
+		if exists {
+			cachedCrd = obj.(*extv1beta1.CustomResourceDefinition)
+		}
+
+		injectOperatorMetadata(kv, &crd.ObjectMeta, version, imageRegistry, id)
+		if exists && objectMatchesVersion(&cachedCrd.ObjectMeta, version, imageRegistry, id) {
+			// Patch if old version
```

Comment:
comment should read, "Patch if new version though", right. because in this case you're only patch if the cached crd matches the current install strategy.